Add some documentation/summaries to meteor type defs

ToastHawaii opened the pull request:
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.meteor.com/#/full/, https://github.com/meteor/meteor/
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

radekmie commented on types/meteor/tsconfig.json:
```diff
@@ -27,4 +27,4 @@
         "test/meteor-mongo-tests.ts"
     ]
-}
+}
```
I'd revert this change for two reasons:
1. We'll need to go through a much longer review to merge this PR.
2. All other files have an additional new line at the end.

radekmie commented on types/meteor/check.d.ts:
```diff
@@ -1,45 +1,84 @@
+            | [Pattern]
+            | { [key: string]: Pattern }
+            | Matcher<any>;
+        export type PatternMatch<T extends Pattern> = T extends Matcher<infer U>
+            ? U
+            : T extends typeof String
+            ? string
+            : T extends typeof Number
+            ? number
+            : T extends typeof Boolean
+            ? boolean
+            : T extends typeof Object
+            ? object
+            : T extends typeof Function
+            ? Function
+            : T extends undefined | null | string | number | boolean
+            ? T
+            : T extends new (...args: any[]) => infer U
+            ? U
+            : T extends [Pattern]
+            ? PatternMatch<T[0]>[]
+            : T extends { [key: string]: Pattern }
+            ? { [K in keyof T]: PatternMatch<T[K]> }
+            : unknown;
```
In my subjective opinion, the previous version was much better. I know it doesn't follow any strict formatting rules (I guess you've just used ESLint, Prettier, or both), but in this very case, I'd keep these in a one-line-per-branch manner.

radekmie commented on types/meteor/accounts-base.d.ts:
```diff
@@ -54,15 +57,29 @@ declare module "meteor/accounts-base" {
+        ): void;
+
+        function forgotPassword(
+            options: {
+                email?: string;
+            },
```
In a couple of places, the formatting expands small and simple objects like this one into multiple ones, just because it was a multi-line object before. How about wrapping them back into one line? It'd shave off a couple of lines from the files and make them _slightly_ more comprehensible.

radekmie commented on types/meteor/meteor.d.ts:
```diff
@@ -64,15 +106,32 @@ declare module "meteor/meteor" {
+             * Set the logged in user.
+             * @param userId The value that should be returned by `userId` on this connection.
+             */
             setUserId(userId: string): void;
```
You've corrected the `ddp.d.ts` file to allow `null`s. I guess applies here as well?

radekmie commented on types/meteor/mongo.d.ts:
```diff
@@ -1,200 +1,338 @@
+            | 'string'
+            | 3
+            | 'object'
+            | 4
+            | 'array'
+            | 5
+            | 'binData'
+            | 6
+            | 'undefined'
+            | 7
+            | 'objectId'
+            | 8
+            | 'bool'
+            | 9
+            | 'date'
+            | 10
+            | 'null'
+            | 11
+            | 'regex'
+            | 12
+            | 'dbPointer'
+            | 13
+            | 'javascript'
+            | 14
+            | 'symbol'
+            | 15
+            | 'javascriptWithScope'
+            | 16
+            | 'int'
+            | 17
+            | 'timestamp'
+            | 18
+            | 'long'
+            | 19
+            | 'decimal'
+            | -1
+            | 'minKey'
+            | 127
+            | 'maxKey'
+            | 'number';
```
Same as in `check.d.ts`, I'd keep the old formatting here. It clearly presented that the string values were aliases of the numerical ones.

ToastHawaii commented:
@radekmie Thank you for your review, I hope I have addressed all your concerns.

About the formatting: I have used prettier as described here https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes

I added some `// prettier-ignore` to prevent some lines from being formatted.

radekmie commented on types/meteor/globals/mongo.d.ts:
```diff
@@ -239,3 +390,14 @@ declare module Mongo {
     }
 }
+
+declare interface MongoConnection {
+    db: any;
+    client: any;
+}
```
Why `any`? It works correctly in `types/meteor/mongo.d.ts` with proper types:
```ts
    interface MongoConnection {
        db: MongoDb;
        client: MongoClient;
    }
```

ToastHawaii replied:
I tried it with `/// <reference types="mongodb" />`
but it didn't work so I described it as `any;` to get further.

radekmie reviewed (commented):
Prettier will never wrap an object back into one line. For example, this code will be preserved as it is:
```ts
type A = { a: string };
type B = {
  b: string;
};
```

And as there are quite a few objects in these typings, it'd get rid of a lot of lines.

ToastHawaii commented:
> Prettier will never wrap an object back into one line. For example, this code will be preserved as it is:
> 
> ```ts
> type A = { a: string };
> type B = {
>   b: string;
> };
> ```
> 
> And as there are quite a few objects in these typings, it'd get rid of a lot of lines.

@radekmie Thank you for your explanation. Now I understand how prettier wrapps.

I have reduced types with only one property to one line if they do not have a long description.

ToastHawaii commented:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54414
